src/engine/renderer/tr_main.cpp

```diff
@@ -1340,45 +1325,101 @@ static bool R_GetPortalOrientations( drawSurf_t *drawSurf, orientation_t *surfac
 			VectorCopy( surface->axis[ 1 ], camera->axis[ 1 ] );
 			VectorCopy( surface->axis[ 2 ], camera->axis[ 2 ] );
 
+			R_MirrorPoint( tr.viewParms.orientation.origin, surface, camera, *outOrigin );
+			VectorAdd( e->e.oldorigin, plane.normal, pvsOrigin );
+
+			VectorMA( tr.viewParms.orientation.axis[0], -2 * DotProduct( tr.viewParms.orientation.axis[0], plane.normal ), plane.normal, outAxis[0] );
+			VectorMA( tr.viewParms.orientation.axis[1], -2 * DotProduct( tr.viewParms.orientation.axis[1], plane.normal ), plane.normal, outAxis[1] );
+			CrossProduct( outAxis[0], outAxis[1], outAxis[2] );
+
 			*mirror = true;
 			return true;
 		}
 
-		// now get the camera origin and orientation
-		VectorCopy( e->e.oldorigin, camera->origin );
-		AxisCopy( e->e.axis, camera->axis );
-		VectorSubtract( vec3_origin, camera->axis[ 0 ], camera->axis[ 0 ] );
-		VectorSubtract( vec3_origin, camera->axis[ 1 ], camera->axis[ 1 ] );
-
 		// optionally rotate
-		if ( e->e.oldframe )
-		{
+		if ( e->e.oldframe ) {
 			// if a speed is specified
-			if ( e->e.frame )
-			{
+			if ( e->e.frame ) {
 				// continuous rotate
 				d = ( tr.refdef.time / 1000.0f ) * e->e.frame;
-				VectorCopy( camera->axis[ 1 ], transformed );
-				RotatePointAroundVector( camera->axis[ 1 ], camera->axis[ 0 ], transformed, d );
-				CrossProduct( camera->axis[ 0 ], camera->axis[ 1 ], camera->axis[ 2 ] );
-			}
-			else
-			{
+				VectorCopy( e->e.axis[ 1 ], transformed);
+				RotatePointAroundVector( e->e.axis[ 1 ], e->e.axis[ 0 ], transformed, d );
+				CrossProduct( e->e.axis[ 0 ], e->e.axis[ 1 ], e->e.axis[ 2 ] );
+			} else {
 				// bobbing rotate, with skinNum being the rotation offset
 				d = sinf( tr.refdef.time * 0.003f );
 				d = e->e.skinNum + d * 4;
```

Comment:
This code still refers to skinNum, but the gamelogic-side PR removes the code from CG_Portal that sets that?

Reply:
Fixed on sgame side.

Reply:
You mean you removed the skinNum usage from the renderer?

Having nuked the rotation offset option, it should be removed from the comment as well.
`// bobbing rotate, with skinNum being the rotation offset`

Reply:
> You mean you removed the skinNum usage from the renderer?

Yes.

> Having nuked the rotation offset option, it should be removed from the comment as well.

Done.